src/attributes.js

```diff
@@ -17,77 +17,123 @@
-
-      for (var prop in style) {
-        el.style[prop] = style[prop];
-      }
-    }
-  },
-
-
-  /**
-   * Updates a single attribute on an Element.
-   * @param {!Element} el
-   * @param {string} name The attribute's name.
-   * @param {*} value The attribute's value. If the value is a string, it is set
-   *     as an HTML attribute, otherwise, it is set on node.
-   */
-  updateAttribute: function(el, name, value) {
-    var data = getData(el);
-    var attrs = data.attrs;
-
-    if (attrs[name] === value) {
-      return;
-    }
+/**
+ * Applies an attribute or property to a given Element. If the value is null
+ * or undefined, it is removed from the Element. Otherwise, the value is set
+ * as an attribute.
+ * @param {!Element} el
+ * @param {string} name The attribute's name.
+ * @param {*} value The attribute's value.
+ */
+var applyAttr = function(el, name, value) {
+  if (value === null || value === undefined) {
+    el.removeAttribute(name);
+  } else {
+    el.setAttribute(name, value);
+  }
+};
+
 
-    if (name === 'style') {
-      attributes.applyStyle(el, value);
-    } else {
-      attributes.applyAttr(el, name, value);
+/**
+ * Applies a property to a given Element.
+ * @param {!Element} el
+ * @param {string} name The property's name.
+ * @param {*} value The property's value.
+ */
+var applyProp = function(el, name, value) {
+  el[name] = value;
+};
+
+
+/**
+ * Applies a style to an Element. No vendor prefix expansion is done for
+ * property names/values.
+ * @param {!Element} el
+ * @param {string} name The attribute's name.
+ * @param {string|Object<string,string>} style The style to set. Either a
+ *     string of css or an object containing property-value pairs.
+ */
+var applyStyle = function(el, name, style) {
+  if (typeof style === 'string') {
+    el.style.cssText = style;
+  } else {
+    el.style.cssText = '';
+
+    for (var prop in style) {
+      el.style[prop] = style[prop];
     }
+  }
+};
+
 
-    attrs[name] = value;
+/**
+ * Updates a single attribute on an Element.
+ * @param {!Element} el
+ * @param {string} name The attribute's name.
+ * @param {*} value The attribute's value. If the value is a string, it is set
+ *     as an HTML attribute, otherwise, it is set on node.
+ */
+var applyAttributeTyped = function(el, name, value) {
+  var type = typeof value;
+
+  if (type === 'object' || type === 'function') {
+    applyProp(el, name, value);
+  } else {
+    applyAttr(el, name, value);
+  }
+};
+
+
+/**
+ * Updates a single attribute on an Element.
+ * @param {!Element} el
+ * @param {string} name The attribute's name.
+ * @param {*} value The attribute's value. If the value is a string, it is set
+ *     as an HTML attribute, otherwise, it is set on node.
+ */
+var updateAttribute = function(el, name, value) {
+  var data = getData(el);
+  var attrs = data.attrs;
+
+  if (attrs[name] === value) {
+    return;
   }
+
+  var mutator = mutators[name] || mutators.all || applyAttributeTyped;
```

Comment:
I don't see a use case for setting mutators.all back to null, you should default mutators.all to applyAttributeTyped